coralproject/talk
# Added authorization pieces + test functions

## What does this PR do?

Adds required authorizations to the api endpoints and adds test harnesses to inject the user into the request.

## How do I test this PR?

Run `npm run test` and see it go! Future tests involving routers must do the following:

```js
const passport = require('../../../utils/passport');
...

return chai.request(app)
      .get('/api/v1/comments')
      .set(passport.inject(<USER OBJECT HERE>))
...
```

Where `<USER OBJECT HERE>` could be `{roles: ['admin']}` to simulate an admin user or `{id: '123', roles: []}` to simulate user with id `123`. Some FE code may have to be updated to support cookies!
